libavformat/dashdec.c

```diff
@@ -1628,7 +1632,7 @@ static int refresh_manifest(AVFormatContext *s)
             // update segments
             int64_t newSeqNo = calc_next_seg_no_from_timelines(c, ccur_video, currentTime * ccur_video->fragment_timescale - 1);
             if (newSeqNo >= 0) {
-                ccur_video->cur_seq_no = newSeqNo;
+                ccur_video->cur_seq_no = FFMAX(newSeqNo, cur_video->cur_seq_no);
```

Comment:
Instead of using `FFMAX` would moving this check up to the if condition also work? What does `move_timelines` do?

So would this be equivalent? 

```
if (newSeqNo >= 0 && newSeqNo > cur_video->cur_seq_no) {
```

I guess also what is `ccur_video->cur_seq_no` set to prior to the if check?

Reply:
At the start of the function, `refresh_manifests()` makes a copy of the current representation state (video, audio, subtitles) before refreshing/parsing the manifest.  This turns into the "working copy".  `parse_manifest` is then called and overwrites the existing state within the `DASHContext` object.

Once the manifest is parsed, it figures out the next sequence number (this chunk of code) and then moves the timeline or segment list back into the "working copy".

Before the function returns, it then updates the `DASHContext` to point at the "working copy" that was made at the start of the function.

It was probably done this way since `parse_manifest` operates on a `DASHContext`.

Reply:
`ccur_video` is a pointer to the latest representation data that was just parsed.  `move_timelines` ends up using it (hence why it needs to be set here) when moving the timeline data that was parsed into the "working copy".

I'm sure this sounds confusing... it would be best to read the function from top to bottom a couple of times.